lib/mayaUsd/ufe/UsdTransform3dMayaXformStack.h

```diff
@@ -0,0 +1,133 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsd/ufe/UsdTransform3dBase.h>
+
+#include <mayaUsd/fileio/utils/xformStack.h>
+
+#include <pxr/usd/usdGeom/xformable.h>
+
+#include <unordered_map>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Transform USD objects in 3D using the Maya transform stack.
+// 
+// The Maya transform stack is described here:
+// http://help.autodesk.com/view/MAYAUL/2018/ENU/?guid=__cpp_ref_class_m_fn_transform_html
+//
+// The Maya transform stack represents a local matrix transformation as a fixed
+// list of transform ops of a prescribed type and semantics.  This class allows
+// for UFE transformation of objects that use this local matrix representation.
+//
+class MAYAUSD_CORE_PUBLIC UsdTransform3dMayaXformStack : public UsdTransform3dBase
+{
+public:
+    enum OpNdx {NdxTranslate = 0, NdxRotatePivotTranslate, NdxRotatePivot, 
```

Comment:
All-in-one index for transform op ordering and rotation transform op equivalence.  The existing import / export Maya transform stack code
https://github.com/Autodesk/maya-usd/blob/f144c43a8df2e6f369c07c28134077b0209b7acf/lib/mayaUsd/fileio/utils/xformStack.cpp#L27
has rotation transform op equivalence, but does not need transform op ordering, as far as I can tell, exporting transform ops to USD in a fixed order, and on import transform op ordering is a given.